app.go

```diff
@@ -474,9 +474,32 @@ func (app *app) loop() {
 	}
 }
 
-type cleanFunc func()
+func (app *app) runCmdSync(cmd *exec.Cmd, pause_after bool) {
+	app.nav.previewChan <- ""
+	app.nav.dirPreviewChan <- nil
 
-var noopCleanFunc = func() {}
+	if err := app.ui.suspend(); err != nil {
+		log.Printf("suspend: %s", err)
+	}
+	defer func() {
+		if err := app.ui.resume(); err != nil {
+			app.quit()
+			os.Exit(3)
+			return
+		}
+	}()
+	defer app.nav.renew()
+	defer app.ui.loadFile(app, true)
```

Comment:
Do you think it would be better to combine these `defer` statements? Also I'm not sure if they are needed since the function doesn't return early anymore and you can just put the cleanup code at the end. This is just a suggestion.

Reply:
I left one of them, in the spirit of putting deferred cleanup next to resource initialization. I'm never sure whether to prefer that or just linear execution.